sources/logdog/src/main.rs

```diff
@@ -0,0 +1,256 @@
+        },
+        // Get Bottlerocket settings using the apiclient.
+        ExecToFile {
+            command: "apiclient",
+            args: vec!["--method", "GET", "--uri", "/settings"],
+            output_filename: "settings.json",
+        },
+        // Get networking status from wicked.
+        ExecToFile {
+            command: "wicked",
+            args: vec!["show", "all"],
+            output_filename: "wicked",
+        },
+        // Get configuration info from containerd.
+        ExecToFile {
+            command: "containerd",
+            args: vec!["config", "dump"],
+            output_filename: "containerd-config",
+        },
+        // Get the status of kubelet and other kube processes from systemctl.
+        ExecToFile {
+            command: "systemctl",
+            args: vec!["status", "kube*", "-l", "--no-pager"],
+            output_filename: "kube-status",
+        },
+        // Get the kernel message buffer with dmesg.
+        ExecToFile {
+            command: "dmesg",
+            args: vec!["--color=never", "--nopager"],
+            output_filename: "dmesg",
+        },
+        // Get firewall filtering information from iptables.
+        ExecToFile {
+            command: "iptables",
+            args: vec!["-nvL", "-t", "filter"],
+            output_filename: "iptables-filter",
+        },
+        // Get firewall nat information from iptables.
+        ExecToFile {
+            command: "iptables",
+            args: vec!["-nvL", "-t", "nat"],
+            output_filename: "iptables-nat",
+        },
+        // Get disk and filesytem information from df.
+        ExecToFile {
+            command: "df",
+            args: vec!["-h"],
+            output_filename: "df",
+        },
+    ]
+}
+
+fn main() -> ! {
+    let output = parse_args(env::args());
+    process::exit(match run_program(&output) {
+        Ok(()) => 0,
+        Err(err) => {
+            eprintln!("{}", err);
+            if let Some(var) = env::var_os("RUST_BACKTRACE") {
+                if var != "0" {
+                    if let Some(backtrace) = err.backtrace() {
+                        eprintln!("\n{:?}", backtrace);
+                    }
+                }
+            }
+            1
+        }
+    })
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use flate2::read::GzDecoder;
+    use std::fs::File;
+    use tar::Archive;
+    use uuid::Uuid;
+
+    #[test]
+    fn test_program() {
+        let output_tempdir =
+            TempDir::new(std::env::temp_dir().join(Uuid::new_v4().to_string())).unwrap();
+        let output_filepath = output_tempdir.path().join("logstest");
+
+        // This should work on any system, even if the underlying programs being called are absent.
+        run_program(&output_filepath).unwrap();
+
+        // Open the file and spot check that a couple of expected files exist inside it.
+        // This function will panic if the path is not found in the tarball
```

Comment:
I can almost guarantee this is going to fail on someone's system, and not just from missing commands, but commands that they would not want to run on their build system... I don't think this test is really safe or necessary.